### PR TITLE
add time_zone into ConfigOptions

### LINKED ISSUE #3255: Add support for `get` / `show` timezone
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

Presently, `show timezone` does not work:

```
DataFusion CLI v11.0.0
❯ show time zone;
NotImplemented(“SHOW time.zone not implemented. Supported syntax: SHOW <TABLES>“)
```

**Describe the solution you'd like**

Instead, I would like:

```
show timezone;
```

To always return 'utc'.

```
set timezone 'utc';
```

Should succeed.

```
set timezone 'UTC+08`;
```
should return an error.

and
```
set timezone 'US/Mountain`;
```
should return the same error.

This should:
1. make the work @waitingkuo is doing easier by ignoring non-UTC zones for now
2. Stay technically compatible with ANSI SQL & postgres (as long as postgres is set to UTC)
3. allow the option to implement timezones at a future date.

PS:

Do the same for `select current_setting('TIMEZONE')`

**Describe alternatives you've considered**

- Having to violate ANSI SQL
- Having to tackle timezones while implementing `timestamp without timezone`

### PR DESCRIPTION
**Pull request by waitingkuo**
# Which issue does this PR close?

Closes #3255 

 # Rationale for this change

some of the function requires session timezone to work
e.g. `EXTRACT(HOUR FROM SOME_TIME)` need this 

as `SHOW` statement is done in #3455 
we could do this now

```bash
❯ show datafusion.execution.time_zone;
datafusion.execution.time_zone
+--------------------------------+---------+
| name                           | setting |
+--------------------------------+---------+
| datafusion.execution.time_zone | UTC     |
+--------------------------------+---------+
1 row in set. Query took 0.004 seconds.
```
I also added the shortcut to align with postgrseql

```bash
❯ show timezone;
timezone
+--------------------------------+---------+
| name                           | setting |
+--------------------------------+---------+
| datafusion.execution.time_zone | UTC     |
+--------------------------------+---------+
1 row in set. Query took 0.004 seconds.

❯ show time zone;
time.zone
+--------------------------------+---------+
| name                           | setting |
+--------------------------------+---------+
| datafusion.execution.time_zone | UTC     |
+--------------------------------+---------+
1 row in set. Query took 0.004 seconds.
```

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by waitingkuo** [comment]
@avantgardnerio  @liukun4515

File: datafusion/core/src/config.rs [hunk 1]
```diff
@@ -102,6 +105,20 @@ impl ConfigDefinition {
...
+        default_value: String,
+    ) -> Self {
+        Self::new(
+            key,
+            description,
+            DataType::UInt64,
+            ScalarValue::Utf8(Some(default_value)),
+        )
+    }
 }
```
Comment (waitingkuo) [inline datafusion/core/src/config.rs:122]: add a `new_string` method for adding String Type configuration

File: datafusion/sql/src/planner.rs [hunk 2]
```diff
@@ -2386,8 +2386,14 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
...
 
-        let query = if variable.to_lowercase() == "all" {
+        let variable_lower = variable.to_lowercase();
+        println!("{}", variable_lower);
+
+        let query = if variable_lower == "all" {
             String::from("SELECT name, setting FROM information_schema.df_settings")
+        } else if variable_lower == "timezone" || variable_lower == "time.zone" {
+            // we could introduce alias in OptionDefinition if this string matching thing grows
+            String::from("SELECT name, setting FROM information_schema.df_settings WHERE name = 'datafusion.execution.time_zone'")
```
Comment (waitingkuo) [inline datafusion/sql/src/planner.rs:2396]: if we have more and more alias like this, we could consider add aliases into `ConfigDefinition`

File: datafusion/sql/src/planner.rs [hunk 3]
```diff
@@ -2386,8 +2386,14 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             ));
         }
 
-        let query = if variable.to_lowercase() == "all" {
+        let variable_lower = variable.to_lowercase();
+        println!("{}", variable_lower);
```
Comment (alamb) [inline datafusion/sql/src/planner.rs:2390]: Perhaps is this a left over debugging `println!`?

**Review by alamb — APPROVED** [review]
Other than the errant `println!` this looks good to me -- thanks @waitingkuo

File: datafusion/sql/src/planner.rs (hunk 3 above)
Comment (waitingkuo) [reply datafusion/sql/src/planner.rs:2390]: my fault, let me push the fix

Comment (waitingkuo) [reply datafusion/sql/src/planner.rs]: @alamb  fixed

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @waitingkuo

File: datafusion/core/src/config.rs [hunk 4]
```diff
@@ -102,6 +105,20 @@ impl ConfigDefinition {
...
+    /// Create a configuration option definition with a string value
+    pub fn new_string(
+        key: impl Into<String>,
+        description: impl Into<String>,
+        default_value: String,
+    ) -> Self {
+        Self::new(
+            key,
+            description,
+            DataType::UInt64,
```
Comment (liukun4515) [inline datafusion/core/src/config.rs:118]:
```suggestion
            DataType::Utf8,
```
I think the data type should be UTF8

Comment (liukun4515) [reply datafusion/core/src/config.rs:118]: cc @waitingkuo

Comment (liukun4515) [reply datafusion/core/src/config.rs:118]: LGTM just this minor comments

**Comment by waitingkuo** [comment]
thank you @liukun4515

**Pull request merged**